yt/frontends/ytdata/utilities.py

```diff
@@ -103,7 +103,12 @@ def save_as_dataset(ds, filename, data, field_types=None, extra_attrs=None):
         _yt_array_hdf5_attr(fh, "unit_registry_json", ds.unit_registry.to_json())
 
     if hasattr(ds, "unit_system"):
-        _yt_array_hdf5_attr(fh, "unit_system_name", ds.unit_system.name.split("_")[0])
+        if hasattr(ds, "_unit_system_name"):
+            _yt_array_hdf5_attr(fh, "unit_system_name", ds._unit_system_name)
+        else:
+            _yt_array_hdf5_attr(
+                fh, "unit_system_name", ds.unit_system.name.split("_")[0]
+            )
```

Comment:
Maybe something like this? Note, I haven't tested if it solves the problem, just a code suggestion.




```suggestion
        name = getattr(ds, "_unit_system_name", ds.unit_system.name.split("_")[0])
        _yt_array_hdf5_attr(fh, "unit_system_name", name)
```

Reply:
After the digging this morning, I thinkg it might also be totally sufficient only write the `_unit_system_name` attribute

```python
_yt_array_hdf5_attr(fh, "unit_system_name", ds._unit_system_name)
```
I initially wasn't 100% sure that having `unit_system` ensures always having `_unit_system_name`, but I suspect it might be the case. I'll check on that more thoroughly...

Reply:
I ended up changing how the unit `unit_system.name` is initially set